piksi_tools/interpolate_event_positions.py

```diff
@@ -0,0 +1,150 @@
+            triggertow = msg.tow
+            dataout = False
+
+        if msg.__class__.__name__ == msgtype and dataout == False :
+          if msgtype == "MsgBaselineECEF" or msgtype == "MsgPosECEF" :
+            if previous_msg.flags == msg.flags: #< interpolates only if lock type didn't change.
+              trigger_x = lin_interp(previous_msg.x, msg.x, previous_msg.tow, msg.tow, triggertow)
+              trigger_y = lin_interp(previous_msg.y, msg.y, previous_msg.tow,msg.tow, triggertow)
+              trigger_z = lin_interp(previous_msg.z, msg.z, previous_msg.tow,msg.tow, triggertow)
+              writer.writerow((triggertow, trigger_x, trigger_y, trigger_z, previous_msg.accuracy,
+                              previous_msg.n_sats, msg.flags))
+            else: #< otherwise outputs previous data packet received.
+              writer.writerow((triggertow, previous_msg.x, previous_msg.y, previous_msg.z,
+                              previous_msg.accuracy, previous_msg.n_sats, previous_msg.flags))
+          elif msgtype == "MsgBaselineNED" :
+            if previous_msg.flags == msg.flags: #< interpolates only if lock type didn't change.
+              trigger_n = lin_interp(previous_msg.n, msg.n, previous_msg.tow,msg.tow, triggertow)
+              trigger_e = lin_interp(previous_msg.e, msg.e, previous_msg.tow,msg.tow, triggertow)
+              trigger_d = lin_interp(previous_msg.d, msg.d, previous_msg.tow,msg.tow, triggertow)
+              writer.writerow((triggertow,trigger_n, trigger_e, trigger_d, previous_msg.h_accuracy,
+                                previous_msg.v_accuracy, previous_msg.n_sats, msg.flags))
+            else: #< otherwise outputs previous data packet received.
+              writer.writerow((triggertow, previous_msg.n, previous_msg.e, previous_msg.d, previous_msg.h_accuracy,
+                                previous_msg.v_accuracy, previous_msg.n_sats, previous_msg.flags))
+          elif msgtype == "MsgPosLLH" :
+            if previous_msg.flags == msg.flags: #< interpolates only if lock type didn't change.
+              trigger_lat = lin_interp(previous_msg.lat ,msg.lat, previous_msg.tow, msg.tow, triggertow)
+              trigger_lon = lin_interp(previous_msg.lon ,msg.lon, previous_msg.tow, msg.tow, triggertow)
+              trigger_height = lin_interp(previous_msg.height, msg.height, previous_msg.tow, msg.tow, triggertow)
+              writer.writerow((triggertow, trigger_lat, trigger_lon, trigger_height, previous_msg.h_accuracy,
+                                previous_msg.v_accuracy, previous_msg.n_sats, msg.flags))
+            else: #< otherwise outputs previous data packet received.
+              writer.writerow((triggertow, previous_msg.lat, previous_msg.lon, previous_msg.height, previous_msg.h_accuracy,
+                                previous_msg.v_accuracy, previous_msg.n_sats, previous_msg.flags))
+          dataout = True
+        if msg.__class__.__name__ == msgtype :
```

Comment:
please add a comment that says you store the last message of the msgtype of interest at all times to be used for interpolation.